projects/test-app/src/hslayers-app/hslayers-app.component.ts

```diff
@@ -37,18 +44,82 @@ export class HslayersAppComponent {
         },
         sidebarPosition: 'right',
       },
-      {
-        name: 'app-2',
-        panelsEnabled: {
-          compositionLoadingProgress: true,
-          tripPlanner: true,
-          mapSwipe: false,
-          feature_table: true,
-        },
-        sidebarPosition: 'left',
-      },
+      // {
+      //   name: 'app-2',
+      //   panelsEnabled: {
+      //     compositionLoadingProgress: true,
+      //     tripPlanner: true,
+      //     mapSwipe: false,
+      //     feature_table: true,
+      //   },
+      //   sidebarPosition: 'left',
+      // },
     ];
     for (const app of apps) {
+      const interpolatedSource = new InterpolatedSource({
+        features: [],
+        weight: 'fac2020',
+        loader: async ({extent, projection}) => {
+          if (this.timer !== null) {
+            clearTimeout(this.timer);
+          }
+          interpolatedSource.cancelUrlRequest.next();
+          const extentIn4326 = transformExtent(extent, projection, 'EPSG:4326');
+          const url = this.hsUtilsService.proxify(
+            interpolatedSource.createIDWSourceUrl(
+              'https://api-agroclimatic.lesprojekt.cz/area/selection/preci/0/{minY}/{maxY}/{minX}/{maxX}/100/random/year/2020/2020/1/5/2020-01-01/2020-01-30/1/1/ERA5-Land',
+              extentIn4326
+            ),
+            app.name
+          );
+          try {
+            const response: any = await lastValueFrom(
+              this.httpClient.get(url).pipe(
+                takeUntil(interpolatedSource.cancelUrlRequest),
+                catchError(async (e) => {})
+              )
+            );
+            if (response?.features?.length > 0) {
+              return interpolatedSource.parseFeatures(response, projection);
+            }
+          } catch (error) {}
+        },
+        setCustomColor: (v) => {
+          let color = [0, 0, 0, 255];
+          if (isNaN(v)) {
+            return color;
+          }
+          v = Math.floor(v);
+          switch (true) {
+            case v < 20:
+              color = interpolatedSource.jetColorMap[10];
+              break;
+            case v < 40:
+              color = interpolatedSource.jetColorMap[30];
+              break;
+            case v < 60:
+              color = interpolatedSource.jetColorMap[50];
+              break;
+            case v < 80:
+              color = interpolatedSource.jetColorMap[70];
+              break;
+            case v <= 100:
+            case v > 100:
+              color = interpolatedSource.jetColorMap[90];
+              break;
+            default:
+              color = [0, 0, 0, 255];
+          }
+          return color;
+        },
+      });
+      const idwLayer = new ImageLayer({
+        properties: {title: 'IDW layer'},
+        source: interpolatedSource as any,
+        opacity: 0.5,
+      });
+      //Manditory, otherwise nothing will be loaded with source loader
```

Comment:
But it is strange, since the loader is set on interpolatedSource. If its mandatory, it should be created from inside interpolatedSource class, but I still have my doubts